File: a1_Marshall/python/benchmark.py
#Un-comment print lines to verify sorting function works
import a1
import sys
import helper

if(sys.argv[1] == None):
	print "File usage: <stringlen> <arraylen>"

stringlen = int(sys.argv[1])
arraylen = int(sys.argv[2])

helper.resetTimer()
arr = helper.makeArray(stringlen, arraylen)
time = helper.readTimerMilliseconds()
#print "\n".join(arr)
print "------------>Took: %d ms to construct" %time

helper.resetTimer()
a1.quicksort(arr)
time = helper.readTimerMilliseconds()
#print "\n".join(arr)
print "------------>Took: %d ms to sort" %time

File: a1_Marshall/go/benchmark.go
package main

import(
	"fmt"
	"os"
	"helper"
	"strconv"
	"quicksort"
)

func main() {
	if(os.Args[1] == ""){
		fmt.Println("File usage: <stringlen> <arraylen>")
		return
	}

	t1 := os.Args[1]
	t2 := os.Args[2]

	stringlen, _ := strconv.Atoi(t1)
	arraylen, _ := strconv.Atoi(t2)

	helper.ResetTimer()
	var arr []string = helper.MakeArray(stringlen, arraylen)
	var time  = helper.ReadTimerMilliseconds()
	fmt.Printf("Generated %d strings of length %d in %.4f ms\n", arraylen, stringlen, time)
/*    	for _, x := range(arr) {
       	fmt.Println(x)
    	} */

    	helper.ResetTimer()
    	quicksort.Sort(arr, 0, arraylen-1)
    	time  = helper.ReadTimerMilliseconds()
			fmt.Printf("Sorted %d strings of length %d in %.4f ms\n", arraylen, stringlen, time)
/*    	for _, x := range(arr) {
       	fmt.Println(x)
    	} */

}
